swig/CMakeLists.txt

```diff
@@ -20,27 +20,47 @@ endif()
 
 # find Python package
 if(GEN_PYTHON_BINDINGS AND SWIG_FOUND)
-    message(STATUS "Python version ${GEN_PYTHON_VERSION} was selected")
-    unset(PYTHON_LIBRARY CACHE)
-    unset(PYTHON_EXECUTABLE CACHE)
-    unset(PYTHON_INCLUDE_DIR CACHE)
-    unset(PYTHON_LIBRARY_DEBUG CACHE)
-    if(${GEN_PYTHON_VERSION} STREQUAL "2")
-        find_package(PythonLibs 2 REQUIRED)
-        find_package(PythonInterp 2 REQUIRED)
-        if(NOT PYTHONLIBS_FOUND)
-            message(WARNING "Did not found Python version 2.x")
-            message(STATUS "Sysrepo supports Python 2.x and Python 3.x")
-        endif()
-    elseif(${GEN_PYTHON_VERSION} STREQUAL "3")
-        find_package(PythonLibs 3 REQUIRED)
-        find_package(PythonInterp 3 REQUIRED)
-        if(NOT PYTHONLIBS_FOUND)
-            message(WARNING "Did not found Python version 3.x")
-            message(STATUS "Sysrepo supports Python 2.x and Python 3.x")
-        endif()
+    if(ENABLE_STATIC)
+        message(WARNING "Can't create a static Python module")
     else()
-        message(WARNING "Sysrepo supports Python 2.x and Python 3.x")
+        foreach(CUR_PYTHON_VERSION ${GEN_PYTHON_VERSION})
+            unset(PYTHON_EXECUTABLE CACHE)
+            unset(PYTHON_INCLUDE_PATH CACHE)
+            unset(PYTHON_EXT_SUFFIX CACHE)
+            unset(PYTHON_MODULE_PATH CACHE)
+
+            # find_package(Python...) does not do the right thing here.  we
```

Comment:
I'm aware of the problem... the issue is that when using `find_package`, it gets completely confused when multiple versions are installed.  For example, Debian 10 requires that all Python software is built for `3.6m 3.7m 3.6dm 3.7dm` versions.

With `find_package`:
* there is no support at all for 2 variants of the same version
* in my tests it kept mixing different libraries and interpreter (e.g. 3.7 libs + 3.6 executable)

I'm trying to redo the test with `pythonX.X-config` which is normally only found if the headers are also installed, since it is part of the `-dev` package.

Reply:
thanks, I understand the resons to not use `find_package`, I just wanted to make note that the replacement is not complete now